airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -142,3 +141,18 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
 
     def get_autocommit(self, conn):
         return getattr(conn, 'autocommit_mode', False)
+
+    def run(self, sql, autocommit=False, parameters=None):
+        """
+        Snowflake-connector doesn't allow natively the execution of multiple SQL statements in the same
+        call. So for allowing to pass files or strings with several queries this method is coded,
+        that relies on run from DBApiHook
+        """
+        if isinstance(sql, str):
```

Comment:
ok so you want to run with execute_string if `str` and use dbapi.run otherwise  (e.g. if it is `List[str]`)

it seems reasonable. but another option would be to use snowflake's `split_statements` function

https://github.com/snowflakedb/snowflake-connector-python/blob/master/src/snowflake/connector/util_text.py#L32

what you could do is always split and then you can always dbapi run.  this would save you the type checking and autocommit logic.

but i think you can reasonably go either way.

one thought though... at my prev company we liked to execute sql files.  what we did was first check if `sql` was a file that exists with `Path(sql).exists()` and if so then `sql = Path(sql).read_text()`.  then execute as normal.  you can do with templating but i find this easier.  anyway it's an idea for your consideration if you want to make this a little more generic --- `Union[Path, List[str], str]` where str can be either single statement, multiple statement, or path to sql file.

Reply:
I have modified the code to implement the function split_statements, thanks for the insight